src/qibo/backends/numpy.py

```diff
@@ -641,8 +653,7 @@ def samples_to_decimal(self, samples, nqubits):
         return self.np.matmul(self.to_numpy(samples), qrange)[:, 0]
 
     def calculate_frequencies(self, samples):
-        res, counts = self.np.unique(samples, return_counts=True)
-        res, counts = self.np.array(res), self.np.array(counts)
+        res, counts = np.unique(samples, return_counts=True)
         return collections.Counter({k: v for k, v in zip(res, counts)})
```

Comment:
I know this isn't something that you did, but
```suggestion
        return collections.Counter(zip(res, counts))
```
works.